core/src/main/java/org/eclipse/hono/util/JsonBackedValueObject.java

```diff
@@ -93,17 +93,20 @@
      * @param parent The JSON to get the property value from.
      * @param name The property name.
      * @param defaultValue A default value to return if the property is {@code null}.
+     * @param clazz The target type.
      * @param <T> The type of the property.
-     * @return The property value or the given default value if the property is not set
-     *         or is of an unexpected type.
+     * @return The property value or the given default value if the property is not set or is of an unexpected type.
      * @throws NullPointerException if any of parent or name are {@code null}.
      */
-    @SuppressWarnings("unchecked")
-    protected final <T> T getProperty(final JsonObject parent, final String name, final T defaultValue) {
+    protected static final <T> T getProperty(final JsonObject parent, final String name, final Class<T> clazz,
```

Comment:
is there any advantage with always passing in the type?

Reply:
Yes, the fact that it works. You cannot cast with a type argument.

Reply:
I guess the problem with the existing code is that it never throws a `ClassCastException`, right?

Reply:
Correct. The type information `T` is gone during runtime and is effectively `Object`. Casting to `Object` will always work.

The compiler did complain about that "Type safety: Unchecked cast from Object to T", but that got suppressed by using `@SuppressWarnings("unchecked")`

This is basically the same case as #751 was, which got fixed by #757.

Searching through the code, I found at least one additional location, having the same issue.

Reply:
Thanks for explaining. Given that this seems to be a bug and that it occurs in other places as well, I think we should fix this in a separate commit/PR. Would you like me to take care of that?

Reply:
Fixing this in another PR will cause conflicts in this PR. I don't see any benefits in that.

My proposal would be to merge this PR, including this change here. And follow up with another PR, fixing the other locations.

Reply:
I understand your desire for getting this PR over with. However, personally, I do not find it that easy to review a PR covering 170 files with that amount of changes.
I am therefore trying to offer some help in reducing the scope of this PR by means of pulling out changes that we need to make in any case and which are not strictly imposed by the functionality being added here. Hope that makes sense ...

Reply:
I absolutely understand your concern. But rebasing this PR is quite time consuming as well.

So I am not looking to push approving this PR, but more reducing the effort that it would take to rebase it over and over again.